src/OgGroupAudienceHelper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og\OgGroupAudienceHelper.
+ */
+
+namespace Drupal\og;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Field\FieldException;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+
+/**
+ * OG audience helper methods.
+ */
+class OgGroupAudienceHelper {
+
+  /**
+   * Return TRUE if a field can be used and has not reached maximum values.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity to check the field cardinality for.
+   * @param string $field_name
+   *   The field name to check the cardinality of.
+   *
+   * @return bool
+   *
+   * @throws \Drupal\Core\Field\FieldException
+   */
+  public static function checkFieldCardinality(ContentEntityInterface $entity, $field_name) {
+    $field_definition = $entity->getFieldDefinition($field_name);
+
+    if (!$field_definition) {
+      throw new FieldException(sprintf('No "%s" field found for %s %s entity', $field_name, $entity->bundle(), $entity->getEntityTypeId()));
+    }
+
+    if (!Og::isGroupAudienceField($field_definition)) {
+      throw new FieldException(sprintf('"%s" field on %s %s entity is not an audience field.', $field_name, $entity->bundle(), $entity->getEntityTypeId()));
```

Comment:
Seems like a field related exception. I think we should add OG specific exceptions though - think I mentioned that somewhere before.